frontend/src/views/WheresMyGeneV2/components/HeatMap/index.tsx

```diff
@@ -130,29 +128,25 @@ export default memo(function HeatMap(props: Props): JSX.Element {
             (isAnyTissueLoading(isLoading) && (
               <ChartWrapper
                 top={xAxisHeight}
-                hidden={!isAnyTissueLoading(isLoading)}
+                visible={isLoadingAPI || isAnyTissueLoading(isLoading)}
               >
-                <SkeletonContainer>
-                  {[...Array(totalElementsCount)].map((_, index) => (
-                    <SkeletonWrapper key={index} data-testid="skeleton-wrapper">
-                      {[...Array(sortedGeneNames.length)].map((_, index) => (
-                        <StyledSkeleton
-                          variant="circular"
-                          width={19}
-                          height={19}
-                          key={index}
-                          data-testid="skeleton-circle"
-                        />
-                      ))}
-                    </SkeletonWrapper>
-                  ))}
-                </SkeletonContainer>
+                <LoadingContainer
+                  height={totalElementsCount}
+                  width={sortedGeneNames.length}
+                >
+                  <LoadingWrapper left={sortedGeneNames.length}>
+                    <LoadingSpinner />
+                    <LoadingLabel visible={sortedGeneNames.length > 12}>
```

Comment:
how did we decided this magic number 12? Can we extract this as a constant and add a comment for context? Thank you!